### PR TITLE
Adding global store access to SmartStore plugin

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by khawkins** [comment]
Seems good.

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js [hunk 1]
```diff
@@ -92,6 +91,7 @@ var StoreCursor = function () {
 
 // ====== Logging support ======
 var logLevel;
+var isGlobalStore;
```
Comment (wmathurin) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js]: logLevel (above) and storeConsole (below) are both for logging support (the previous comment) - so maybe isGlobaStore should be declared before the Logging support comment?

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js [hunk 2]
```diff
@@ -102,13 +102,24 @@ var setLogLevel = function(level) {
...
 var getLogLevel = function () {
     return logLevel;
 };
 
+var setIsGlobal = function(global) {
+    isGlobalStore = global;
+};
+
+// Setting global as false by default.
+setIsGlobal(false);
```
Comment (wmathurin) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js]: That could also be achieved when you are declaring isGlobalStore

**Comment by wmathurin** [comment]
Looks good otherwise.

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js (hunk 1 above)
Comment (bhariharan) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js]: Will do.

**Comment by wmathurin** [comment]
After some thiniking, I'm not sure the approach will work.

For instance, if you use smartsync.js, you will have models/collections backed by various ForceCache objects (ForceCache is a convenient object to read/write to/from a soup in the smartstore). So If a need a collection going to the global smartstore and another one going to the user smartstore, I'll need to remember to flip the switch setIsGlobal back and forth before any save or fetch operations. And what if they overlap??

Maybe we should add a global parameter as a last (and therefore optional) argument to all smartstore methods (the ones that actually do a cordova exec call across the bridge and need the global flag). It would be backward compatible (not passing the flag would be the same as passing false).
Inside each methods, we would make sure to do:
 global = global || false;
to turn no value into the boolean false.

**Comment by khawkins** [comment]
Ah yes, I didn't think about those implications.  That seems like a good way to solve the problem: make the "isGlobal" configuration about the call, rather than object-wide.

**Comment by wmathurin** [comment]
And actually, to be able to use smartsync.js with a global smartstore, we will have to enhance the ForceCache object to not only capture the target soup but also whether or not it lives in the global store or not.

**Comment by bhariharan** [comment]
@wmathurin I've refactored things to pass in the optional argument to the ```SmartStore``` functions.

**Comment by wmathurin** [comment]
:+1:

**Pull request merged**